doc/contributing/contributing.rst

```diff
@@ -31,49 +31,55 @@ To contribute to the source code of the model you will need to fork the reposito
 Quickstart Guide
 ----------------
 
-1. Fork the project on GitHub (using the fork button) and create a local clone, if you haven’t already:
+**1.** Fork the project on GitHub (using the fork button).
+
+**2.** Create a local clone (we strongly suggest keeping a separate repository for development work):
 
 ::
 
     % git clone https://github.com/user_name/MITgcm.git
 
-2. Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
+**3.** Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
 
 ::
 
     % git remote add upstream https://github.com/altMITgcm/MITgcm.git
 
-3. This step is optional, but highly recommended. It will ensure that your GitHub repo stays up to date with the main repository. Checkout the master branch and sync everywhere:
+**4.** Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a new feature) and make edits on this branch:
 
 ::
 
-   % git checkout master
-   % git pull upstream master
-   % git push origin master
-   
-4. Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a bugfix) and make edits on this branch:
+   % git fetch upstream
+   % git checkout -b newfeature upstream/master
 
-::
+**5.** When edits are done, do all git add’s and git commit’s. In the commit, make a succinct (<70 char) summary, followed by a blank line and a longer description. Reference any outstanding issues addressed using the syntax ``#ISSUE_NUMBER``.
 
-   % git fetch upstream
-   % git checkout -b bugfix upstream/master
+**6.** Push the edited branch to the origin remote (i.e. your fork) on GitHub:
 
-5. When edits are done, do all git add’s and git commit’s. In the commit, make a succinct (<70 char) summary, followed by a blank line and a longer description. Reference any outstanding issues addressed using the syntax ``#ISSUE_NUMBER``.
+::
 
+    % git push -u origin newfeature
 
-6. Push the edited branch to the origin remote (i.e. your fork) on GitHub:
+**7.** On GitHub, go to your fork and hit the pull request (PR) button, and wait for the MITgcm head developers to review your proposed changes. You may get additional questions or requests before your changes are accepted into the primary MITgcm source code.
```

Comment:
Should we give explicit instructions on how to update a PR in response to a request for changes? I think so.

- Possibly update your feature branch if master has changed significantly (see instructions below) 
- Edit your `newfeature` branch locally
- `git push origin newfeature`
- Your PR will automatically be updated

Reply:
We do give more detailed instructions, but not in the quickstart guide, since we assume people reading that are somewhat familiar with git and GitHub. Step 7 of the longer version includes the following:
"While the PR remains open, you can go back to step #5 and make additional edits, git adds, git commits, and then redo step #6; such changes will be added to the PR (and maintainers re-notified), no need to redo step #7."